dbms/src/Server/HTTPHandler.cpp

```diff
@@ -445,19 +447,20 @@ void HTTPHandler::processQuery(
         else
         {
             /// All other query parameters are treated as settings.
+            String value;
+            if (!settings.tryGet(it->first, value) || it->second != value)
+            {
```

Comment:
And you forgot about all other places where settings are possible:
- SET query;
- settings passed with query to TCPHandler;
- SETTINGS clause in SELECT.